extensions/amp-story/1.0/amp-story.js

```diff
@@ -2237,13 +2240,27 @@ export class AmpStory extends AMP.BaseElement {
 
     const pagesByDistance = this.getPagesByDistance_();
 
-    this.mutateElement(() => {
+    const preloadAllPages = () => {
       pagesByDistance.forEach((pageIds, distance) => {
         pageIds.forEach((pageId) => {
           const page = this.getPageById(pageId);
           page.setDistance(distance);
         });
       });
+    };
+
+    this.mutateElement(() => {
+      if (prioritizeActivePage) {
+        // Load page with distance 0 first, and then load the other ones.
+        return Promise.all(
+          pagesByDistance[0].map((pageId) => {
+            const page = this.getPageById(pageId);
+            page.setDistance(0);
+            return listenOncePromise(page.element, EventType.PAGE_LOADED);
```

Comment:
- What if the page is already loaded: plz switch to `LOAD_END` signal which covers this case
- What if the user navigates before the active page is loaded? If the code eventually runs it'll override the correct distance with the previous one